### PR TITLE
Report type errors with full type names (fixes #1082)

### LINKED ISSUE #1082: Shadowed binding results in confusing type mismatch error
I ran into a type mismatch that was confusing because the reported expected and actual types were identical. The difference turned out to be that I had (intentionally) shadowed a variable with a formal parameter in a let binding, and my types differed in which binding was referenced. Here's a reduced example of the error message:

```
module Shadowed_type_mismatch

open FStar.Seq

val index' : ns:seq nat -> i:nat{i < length ns} -> nat
let index' ns i =
  let f (ns:seq nat) : nat =
      index ns i in
  f ns
```

```
$ fstar.exe shadowed_type_mismatch.fst
./shadowed_type_mismatch.fst(8,15-8,16): (Error) Subtyping check failed; expected type (i:Prims.nat{ Prims.b2t (i < FStar.Seq.Base.length u#0 ns) }); got type (i:Prims.nat{ Prims.b2t (i < FStar.Seq.Base.length u#0 ns) }) (see also /usr/local/Cellar/fstar/HEAD-cf96c5a_1/ulib/FStar.Seq.Base.fst(30,49-30,61))
Verified module: Shadowed_type_mismatch (583 milliseconds)
1 error was reported (see above)
```

It would be great if these two bindings were disambiguated somehow.

### PR DESCRIPTION
**Pull request by artagnon**
Include a NFC commit to strip commented out code in a related file. (Fixes #1082)

### CODE REVIEW

**Comment by mtzguido** [comment]
I don't think this is an improvement. Firstly we only intend to use `--ugly` for debugging the compiler, since the pretty printer is not yet optimal (as you notice here). Also, the original error does already differentiate `ns2258` from `ns2224` (the problem is not the `iXXXX`: those are binders).

I think a proper solution would be to disambiguate these by turning one of them into `ns0`, `ns'` or something similar. That involves some work in the pretty printer to rename shadowed variables.

As for the removal of commented code, I tend to agree with it.

File: src/syntax/FStar.Syntax.Print.fs [hunk 1]
```diff
@@ -702,11 +656,10 @@ let rec sigelt_to_string (x: sigelt) =
         else U.format4 "%stype %s %s : %s" quals_str lid.str binders_str term_str
       | Sig_datacon(lid, univs, t, _, _, _) ->
         if (Options.print_universes())
-        then //let univs, t = Subst.open_univ_vars univs t in (* AR: don't open the universes, else it's a bit confusing *)
+        then
```
Comment (nikswamy) [inline src/syntax/FStar.Syntax.Print.fs]: Retaining the AR comment here is useful.

File: src/typechecker/FStar.TypeChecker.Err.fs [hunk 2]
```diff
@@ -67,7 +67,7 @@ let err_msg_type_strings env t1 t2 :(string * string) =
   let s2 = N.term_to_string env t2 in
   if s1 = s2 then
     Options.with_saved_options (fun _ ->
-      ignore (Options.set_options Options.Set "--print_full_names --print_universes");
+      ignore (Options.set_options Options.Set "--print_full_names --ugly --print_universes");
```
Comment (nikswamy) [inline src/typechecker/FStar.TypeChecker.Err.fs]: Using --ugly here is too aggressive.

**Comment by nikswamy** [comment]
In general, we have too much commented code lying around and it's nice to clean it up.

But, in this case, FStar.Syntax.Print has a lot of functionality specifically intended for debugging, including printers at various levels of detail. I (and others, I think) occasionally resurrect some of this commented code when debugging.

Instead of removing those bits of code altogether, what do you think about retaining it as code, but in clearly demarcated way, e.g., putting it in functions named debug_*

**Comment by mtzguido** [comment]
I also sometimes resurrect it. Wrapping inside an `if false` is another option.

**Comment by nikswamy** [comment]
> Include a NFC commit to strip commented out code in a related file. (Fixes #1082)

For what it's worth, I had to look up the definition of NFC:   : )
http://lists.llvm.org/pipermail/llvm-commits/Week-of-Mon-20140901/233938.html

**Comment by aseemr** [comment]
Ping @artagnon, would you be up for fixing the review comments? I will be happy to merge it then, thanks!

**Comment by artagnon** [comment]
Frankly, this isn’t a high priority item: I’m busy with other things at the
moment. You can close this for now. Thanks :)

**Pull request closed without merging**